frontend/src/employee-frontend/components/invoice/InvoiceRowsSectionRow.tsx

```diff
@@ -273,4 +253,89 @@ const UnitPriceInput = React.memo(function UnitPriceInput({
   )
 })
 
+type UnitComboboxProps = Pick<Props, 'unitDetails' | 'update'> & {
+  items: UUID[]
+  selectedItem: UUID | null
+}
+
+const UnitCombobox = React.memo(function UnitCombobox({
+  items,
+  selectedItem,
+  unitDetails,
+  update
+}: UnitComboboxProps) {
+  const { i18n } = useTranslation()
+
+  const unitMenuItem = useCallback(
+    ({ item: unitId, highlighted }: MenuItemProps<UUID>) => {
+      const unit = unitDetails[unitId]
+      return (
+        <UnitMenuItem className={classNames({ highlighted, clickable: true })}>
+          <span>{unit?.name ?? unitId}</span>
+          {unit?.costCenter && (
+            <UnitCostCenter>{unit.costCenter}</UnitCostCenter>
+          )}
+        </UnitMenuItem>
+      )
+    },
+    [unitDetails]
+  )
+  const onChange = useCallback(
+    (value: UUID | null) => update({ unitId: value }),
+    [update]
+  )
+  const getItemLabel = useCallback(
+    (unitId: UUID) => unitDetails[unitId]?.name ?? unitId,
+    [unitDetails]
+  )
```

Comment:
I think showing the cost center in the item label could be valuable or was it left out to save space?

Reply:
There's not much space, and based on a quick UX talk it's probably better leave all the space for the name. If we get feedback that the cost center is important enough to be shown there, we can add it easily later.